web/app/scripts/src/services/proportion-field.es.js

```diff
@@ -0,0 +1,61 @@
+/*
+ *
+ * This module calculates the survey completion percentage/
+ *
+ */
+
+import { angular } from '../libraries';
+
+export const moduleProperties = {
+    moduleName: 'incuna-surveys.proportion-field',
+    componentName: 'proportionField'
+};
+
+const module = angular.module(moduleProperties.moduleName, []);
+
+module.service(moduleProperties.componentName, [
+    function () {
+
+        this.buildFields = function (choices, defaults, autoId) {
+            return choices.map((choice, index) => {
+                return Object.assign(
+                    {},
+                    defaults,
+                    {
+                        label: choice,
+                        id: `${autoId}-${index}`
+                    }
+                );
+            });
+        };
+
+        this.calculateTotal = function (values) {
+            return Object.keys(values).reduce((value, key) => {
+                    let current = 0;
+                    if (angular.isNumber(values[key])) {
+                       current = parseInt(values[key], 10)
+                    }
+                    return value + current
+                },
+                0
+            );
```

Comment:
This indentation is getting a bit crazy here, would you mind splitting this up a bit?